dot/core/service.go

```diff
@@ -249,6 +248,13 @@ func (s *Service) handleBlock(block *types.Block, state *rtstorage.TrieState) er
 }
 
 func (s *Service) handleCodeSubstitution(hash common.Hash, state *rtstorage.TrieState) error {
+	return s._handleCodeSubstitution(hash, state, wasmer.NewInstance)
+}
+func (s *Service) _handleCodeSubstitution(
+	hash common.Hash,
+	state *rtstorage.TrieState,
+	newInstance func(code []byte, cfg *wasmer.Config) (*wasmer.Instance, error),
```

Comment:
I'm thinking we should leave as is so that handleCodeSubstitution uses wasmer.NewInstance by default without requiring us to set it manually as a field of Service. If you really think I should change I can, but I think since its only used in the one function and even that's just for testing that its easier to leave as is.

Reply:
I disagree with storing this in `Service`. IMO doesn't make a lot of sense that this is part of the "state" of `Service`.  This is just to make `handleCodeSubsitution` more testable and it should be localised to that function.

Reply:
I get your point although having 2 `handleCodeSubstitution` is kind of ugly too.

Reply:
Conclusion:

- pass it as function argument so it's more explicit and localized
- have 2 similarly-named functions which is not too pretty, but worth the point above!